node_package/src/RSCServerRoot.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { createFromNodeStream } from 'react-on-rails-rsc/client.node';
+import { PassThrough } from 'stream';
+import type { RenderFunction, RailsContext } from './types';
+import transformRSCStream from './transformRSCNodeStream';
+import loadJsonFile from './loadJsonFile';
+import RSCPayloadContainer from './RSCPayloadContainer';
+import { ensureReactUseAvailable } from './reactApis';
+
+ensureReactUseAvailable();
+
+declare global {
+  function generateRSCPayload(
+    componentName: string,
+    props: Record<string, unknown>,
+    railsContext: RailsContext,
+  ): Promise<NodeJS.ReadableStream>;
+}
+
+type RSCServerRootProps = {
+  componentName: string;
+  componentProps: Record<string, unknown>;
+};
+
+const { use } = React;
+
+const createFromReactOnRailsNodeStream = (
+  stream: NodeJS.ReadableStream,
+  ssrManifest: Record<string, unknown>,
+) => {
+  const transformedStream = transformRSCStream(stream);
+  return createFromNodeStream(transformedStream, ssrManifest);
+};
+
+const createSSRManifest = async (
+  reactServerManifestFileName: string,
+  reactClientManifestFileName: string,
+) => {
+  const [reactServerManifest, reactClientManifest] = await Promise.all([
+    loadJsonFile(reactServerManifestFileName) as Promise<Record<string, { id: string; chunks: string[] }>>,
+    loadJsonFile(reactClientManifestFileName) as Promise<Record<string, { id: string }>>,
+  ]);
+
+  const ssrManifest = {
+    // The `moduleLoading` property is utilized by the React runtime to load JavaScript modules.
+    // It can accept options such as `prefix` and `crossOrigin` to specify the path and crossorigin attribute for the modules.
+    // In our case, since the server code is bundled into a single bundle, there is no need to load additional JavaScript modules.
+    // As a result, we set this property to an empty object because it will not be used.
+    moduleLoading: {},
+    moduleMap: {} as Record<string, unknown>,
+  };
+
+  Object.entries(reactClientManifest).forEach(([aboluteFileUrl, clientFileBundlingInfo]) => {
+    const { id, chunks } = reactServerManifest[aboluteFileUrl];
+    ssrManifest.moduleMap[clientFileBundlingInfo.id] = {
+      '*': {
+        id,
+        chunks,
+        name: '*',
+      },
+    };
+  });
```

Comment:
```suggestion
  const moduleMap: Record<string, unknown> = {};

  Object.entries(reactClientManifest).forEach(([aboluteFileUrl, clientFileBundlingInfo]) => {
    const { id, chunks } = reactServerManifest[aboluteFileUrl];
    moduleMap[clientFileBundlingInfo.id] = {
      '*': {
        id,
        chunks,
        name: '*',
      },
    };
  });

  const ssrManifest = {
    // The `moduleLoading` property is utilized by the React runtime to load JavaScript modules.
    // It can accept options such as `prefix` and `crossOrigin` to specify the path and crossorigin attribute for the modules.
    // In our case, since the server code is bundled into a single bundle, there is no need to load additional JavaScript modules.
    // As a result, we set this property to an empty object because it will not be used.
    moduleLoading: {},
    moduleMap,
  };
```
is a bit clearer, I think.

Or we could make
```
const moduleMap = Object.fromEntries(Object.entries(reactClientManifest).map(...)
```
but I think this code ends up less readable.

Reply:
Optionally inline `ssrManifest` after.